modules/navier_stokes/src/materials/INSFVMushyPorousFrictionMaterial.C

```diff
@@ -0,0 +1,66 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "INSFVMushyPorousFrictionMaterial.h"
+#include "MooseMesh.h"
+#include "NS.h"
+
+registerMooseObject("NavierStokesApp", INSFVMushyPorousFrictionMaterial);
+
+InputParameters
+INSFVMushyPorousFrictionMaterial::validParams()
+{
+  InputParameters params = FunctorMaterial::validParams();
+  params.addClassDescription(
+      "Computes the mushy zone porous resistance for solidification/melting problems.");
+  params.addRequiredParam<MooseFunctorName>("liquid_fraction", "Liquid Fraction Functor.");
+  params.addRequiredParam<MooseFunctorName>(NS::mu, "The liquid dynamic viscosity.");
+  params.addRequiredParam<MooseFunctorName>("rho_l", "The liquid density (not the mixture one).");
+  params.addParam<MooseFunctorName>(
+      "dendrite_spacing_scaling", "1e-4", "The dendrite spacing scaling.");
+  return params;
+}
+
+INSFVMushyPorousFrictionMaterial::INSFVMushyPorousFrictionMaterial(
+    const InputParameters & parameters)
+  : FunctorMaterial(parameters),
+    _fl(getFunctor<ADReal>("liquid_fraction")),
+    _mu(getFunctor<ADReal>(NS::mu)),
+    _rho_l(getFunctor<ADReal>("rho_l")),
+    _dendrite_spacing_scaling(getFunctor<ADReal>("dendrite_spacing_scaling"))
+{
+
+  addFunctorProperty<ADReal>(
+      "Darcy_coefficient",
+      [this](const auto & r, const auto & t) -> ADReal
+      {
+        constexpr Real epsilon = 1e-15; // prevents explosion of sqrt(x) derivative to infinity
+
+        const auto fs = 1.0 - _fl(r, t);
+        const auto cs = _c / Utility::pow<2>(_dendrite_spacing_scaling(r, t));
```

Comment:
we should add an assert to avoid division by 0 here. Should be positive in general.

I ll add an issue for the general case

Reply:
I can make the `_dendrite_spacing_scaling` input parameter range checked

Reply:
Never mind it is a functor

Reply:
@lindsayad Is an assert in the lambda going to be thread safe?

Reply:
raised #23463  in the meantime

Reply:
I think from your perspective here, you should regard it as being thread-safe. In practice it may not be, but if it's not, it's not your fault; it would be the framework's fault.

Reply:
In practice I have never seen a thread failure due to a `mooseAssert` but we also never run CIVET thread testing with `devel` or `dbg` modes